// hippo-components/src/components/Carousel/Carousel.js
import React from 'react';
import PropTypes from 'prop-types';
import {
    CarouselProvider,
    Slider,
    Slide,
    ButtonBack,
    ButtonNext,
    DotGroup,
    Dot
} from 'pure-react-carousel';
//Styles
import './Carousel.css';
import 'pure-react-carousel/dist/react-carousel.es.css';
//Icons
import Arrow from './assets/right-arrow.svg';

const Carousel = ({
                      slides,
                      visibleSlides,
                      infinite,
                      className,
                      customArrow,
                      hasMasterSpinner,
                      isPlaying,
                      showDots
                  }) => {
    return (
        <div className={'carousel_container'}>
            <CarouselProvider
                naturalSlideWidth={500}
                naturalSlideHeight={300}
                totalSlides={slides.length}
                visibleSlides={visibleSlides}
                infinite={infinite}
                className={className}
                disableActiveDots={true}
                hasMasterSpinner={hasMasterSpinner}
                isPlaying={isPlaying}
            >
                <Slider
                    className={'carousel_slider_wrapper'}
                >
                    {
                        slides.map((item, index) =>
                            <Slide key={`slide-${index}`} index={index}>
                                {item}
                            </Slide>
                        )
                    }
                </Slider>
                {
                    showDots
                    && <DotGroup className={'carousel_dot_group'} />
                }
                <div className={'carousel_arrows_container'}>
                    <ButtonBack className={'carousel_arrow carousel_back'}>
                        {
                            customArrow
                                ? customArrow
                                : <img alt={'back-arrow'} src={Arrow} />
                        }
                    </ButtonBack>
                    <ButtonNext className={'carousel_arrow carousel_next'}>
                        {
                            customArrow
                                ? customArrow
                                : <img alt={'next-arrow'} src={Arrow} />
                        }
                    </ButtonNext>
                </div>
            </CarouselProvider>
        </div>
    )
};

Carousel.prototypes = {
    slides: PropTypes.array,
    visibleSlides: PropTypes.number,
    infinite: PropTypes.bool,
    hasMasterSpinner: PropTypes.bool,
    isPlaying: PropTypes.bool,
    showDots: PropTypes.bool,
    step: PropTypes.number,
    className: PropTypes.string,
    customArrow: PropTypes.node
};

Carousel.defaultProps = {
    infinite: false,
    hasMasterSpinner: false,
    step: 1,
    isPlaying: false,
    showDots: true,
    className: ''
}

export default Carousel;
